File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int n = nums.size();
        if(n == 0) return {-1, -1};
        int left = 0, right = n-1;
        while(left <= right){
            int mid = left + (right - left)/2;
            if(nums[mid] < target){
                left++;
            }
            else if (nums[mid] > target){
                right--;
            }
            else {
                int l = mid, r = mid;
                for(int i = mid;i>=0;i--){
                    if(nums[i] != target) break;
                    l = i;
                }
                for(int i = mid;i<n;i++){
                    if(nums[i] != target) break;
                    r = i;
                }
                return {l,r};
            }
        }
        return {-1, -1};
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.dart
class Solution {
  List<int> searchRange(List<int> nums, int target) {
    if(nums.length == 0) return [-1, -1];
    List<int> res = [];
    int l = 0, r = nums.length -1;
    while(l <= r){
        int mid = (r + l) ~/ 2;
        if(nums[mid] == target){
            int left=mid, right=mid;
            for(;left>=0 && nums[left]==target;left--);
            for(;right<nums.length && nums[right]==target;right++);
            left++;
            right--;
            return [left, right];
        } else if(nums[mid] > target){
            r = mid-1;
        } else {
            l = mid+1;
        }
    }
    return [-1, -1];
  }
}